src/middlewared/middlewared/plugins/disk.py

```diff
@@ -155,6 +167,10 @@ async def disk_extend_context(self, rows, extra):
         if context['passwords']:
             context['disks_keys'] = await self.middleware.call('kmip.retrieve_sed_disks_keys')
 
+        if context['supports_smart']:
+            if len(rows) > 1:
+                raise CallError('`supports_smart` cannot be queried if disk count is greater than 1')
```

Comment:
Does it make more sense to maybe raise an exception in query method if `supports_smart` is specified without the `get` option?

Don't forget to update docstring :)

Reply:
Idea being that caller is explicitly requesting that they only want one result.

Reply:
@anodos325 but it's perfectly fine to query by `disk_identifier` and expect an empty list as one of the possible return values.

Reply:
Okay. docstring still needs to be updated to reflect new behavior (raises exception if more than one result).